### ISSUE TITLE
compass heading uint16_t overflow

### ISSUE THREAD

**Issue by theg4sh** [PX4/Firmware#10754]
**Describe the bug**
Passing an negative value in parameters to command DO_REPOSITION makes an invalid value of compass_heading after few seconds.

**To Reproduce**
0. Run simulation (probably, the issue could be reproduced in a real life, hadn't check) with, e.g., PX4_HOME_LAT="45.3973425" and PX4_HOME_LON="8.3973425"
1. `rosservice call /mavros/cmd/takeoff "{min_pitch: 0.0, yaw: 0.0, latitude: 0.0, longitude: 0.0, altitude: 505.0}"` <-  Set your AMSL altitude
2. `rosservice call /mavros/cmd/arming "value: true"` Let the drone start to takeoff.
3. Set parameter MIS_YAWMODE to 0 (zero, heading as set by waypoint) in `qgroundcontrol`.
4. `rostopic echo /mavros/home_position/home` and fill `latitude` and `longitude` values in script below.
5. You could use the script to rotate the drone:
```
lat="$(echo "45.3973425*(10^7)" | bc -l)" # change to your lat
lon="$(echo "8.3973425*(10^7)" | bc -l)" # change to your lon
alt=545 # AMSL meters
angle="$(echo "$1/180*3.1415" | bc -l | awk '{printf "%f", $0}')" # degrees
if [ -z "$angle" ]; then
        echo "Angle is not set"
        exit 1
fi
echo "Angle: $1 deg -> $angle rad"
rosservice call /mavros/cmd/command "{broadcast: false, command: 192, confirmation: 0, param1: 0.0, param2: 1.0, param3: 0.0, param4: $angle, param5: $lat, param6: $lon, param7: $alt}"
```
6. Run the script at first time with passing positive argument's value, e.g. '25', and then run with negative argument's value, e.g. '-135'. If you named script as `do_reposition_cmd.sh` then you need to run:
```
bash ./do_reposition_cmd.sh 25
bash ./do_reposition_cmd.sh -135
```
7. `rostopic echo /mavros/global_position/compass_hdg`

**Expected behavior**
In output of command (5) expected the value in range [0.0..360.0), but in that case you'll see the following:
```
...
data: 1.78
---
data: 0.92
---
data: 0.14
---
data: 654.77
---
data: 654.08
---
data: 653.43
```

In `MAVLINK Inspector` (QGC) we observe the same problem: when value reached 0 it jumps to 65535 and continue to decrese.

**Additional context**
https://gist.github.com/theg4sh/9c2508ccda2f4014d2207af6af76692e

UPD:
Somewere, normalized angle with type `int16_t` or `float` with value range  [-180.0..180.0] casting to uint16_t. So, there is two jumps - between 655.35<->0.0 and 179.99<->475.36.

**Comment by theg4sh** [comment]
Workaround in the subscriber's handler could be the following:
```
    float azimuth = msg->data;
    if (azimuth>360.0f) {
        azimuth = azimuth - (std::numeric_limits<uint16_t>::max()+1.0f)/1E2f;
        ROS_DEBUG_THROTTLE(5.0, "Got wrong compass_hdg value: %f, fixed to %f", msg->data, azimuth);
    }
```

**Comment by theg4sh** [comment]
duplicate https://github.com/PX4/Firmware/issues/9867
fixed in v1.8.1
